pkg/cli/debug/debug.go

```diff
@@ -453,20 +456,40 @@ func (o *DebugOptions) RunDebug() error {
 			}
 			return fmt.Errorf(msg)
 			// switch to logging output
-		case err == krun.ErrPodCompleted, err == conditions.ErrContainerTerminated, !o.Attach.Stdin:
-			return logs.LogsOptions{
-				Object: pod,
-				Options: &corev1.PodLogOptions{
-					Container: o.Attach.ContainerName,
-					Follow:    true,
-				},
-				RESTClientGetter: o.RESTClientGetter,
-				ConsumeRequestFn: logs.DefaultConsumeRequest,
-				IOStreams:        o.IOStreams,
-				LogsForObject:    o.LogsForObject,
-			}.RunLogs()
+		case err == krun.ErrPodCompleted, err == conditions.ErrContainerTerminated:
+			o.logsOptions(pod)
+		case err == conditions.ErrNonZeroExitCode:
+			r, w := io.Pipe()
+			streams := genericclioptions.IOStreams{
+				In:     r,
+				Out:    w,
+				ErrOut: ioutil.Discard,
+			}
+			go func() {
+				fmt.Fprintf(w, "%v", logs.LogsOptions{
+					Object: pod,
+					Options: &corev1.PodLogOptions{
+						Container: o.Attach.ContainerName,
+						Follow:    true,
+					},
+					RESTClientGetter: o.RESTClientGetter,
+					ConsumeRequestFn: logs.DefaultConsumeRequest,
+					IOStreams:        streams,
+					LogsForObject:    o.LogsForObject,
+				}.RunLogs())
```

Comment:
Why can't you just re-use `logsOptions` here passing streams accordingly?

Reply:
because you have to capture the output and push to error, otherwise the cmd output will _always_ return 0, even if container exits non-zero.

Reply:
I've added a new getPodLogsForError fn, ptal